selectrum.el

```diff
@@ -478,6 +491,22 @@ Passed to various hook functions.")
                                  candidate
                                  'face 'selectrum-current-candidate)))
               (insert "\n" candidate)
+              (let ((right-margin
+                     (get-text-property
+                      0 'selectrum-candidate-display-right-margin
+                      candidate)))
+                (when right-margin
+                  (let ((ol (make-overlay (point) (point))))
+                    (overlay-put
+                     ol 'after-string
+                     (concat
+                      (propertize
+                       " "
+                       'display
+                       `(space :align-to (- right-fringe
+                                            ,(length right-margin))))
```

Comment:
I noticed that the implementation in lsp-ui-sideline had a hack to account for graphical versus terminal frames. Does your implementation work in both frame types? If so, no need to port the hack -- but just wanted to check.

Reply:
Thanks for pointing this out, it actually has the problem in terminals. See the docstring of `selectrum-right-margin-padding`.